dvc/repo/fetch.py

```diff
@@ -98,6 +99,16 @@ def cb(result):
             root = PathInfo(repo.root_dir)
             for path in files:
                 path_info = root / path
+                try:
+                    used = repo.used_cache(
+                        [os.fspath(path_info)],
+                        force=True,
+                        jobs=jobs,
+                        recursive=True,
+                    )
+                    cb(repo.cloud.pull(used, jobs))
```

Comment:
Strictly speaking this is not really needed, as `cache.save()` will fetch everything anyway, but I've added this for now because logic around this (e.g. tests) expects specific exceptions and also because this might be faster than `save()`, as we didn't optimize it yet.